### PR TITLE
Correct units for normalizing vectors.

### LINKED ISSUE #4192: Unit errors when calling Modelica.Mechanics.MultiBody.Frames.from_nxy
`Modelica.Mechanics.MultiBody.Frames.from_nxy` is expecting unit `"1"` inputs but in at least:
`Modelica.Mechanics.MultiBody.Parts.BodyCylinder` and `Modelica.Mechanics.MultiBody.Parts.BodyBox` it is being called with variables expressed in `"m"`.

I see two solutions to address this issue, one would be to divide the corresponding variable by its `length`, and the other would be to call `Modelica.Units.Conversions.to_unit1`.

Which one is preferred?

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #4192
The basic idea is that non-normalized vectors don't need unit.

### CODE REVIEW

**Comment by HansOlsson** [comment]
Similarly as #4191 I checked number of warnings in Dymola, on master there was 2301 warnings, with this change it decreased to 2124 (with proposed unit-checking).

File: Modelica/Utilities/Internal.mo [hunk 1]
```diff
@@ -21,9 +21,9 @@ package PartialModelicaServices
       "Position vector from origin of world frame to origin of object frame, resolved in world frame" annotation(Dialog);
     input SI.Position r_shape[3]={0,0,0}
       "Position vector from origin of object frame to shape origin, resolved in object frame" annotation(Dialog);
-    input Real lengthDirection[3](each final unit="1")={1,0,0}
+    input Real lengthDirection[3]={1,0,0}
       "Vector in length direction, resolved in object frame" annotation(Dialog);
-    input Real widthDirection[3](each final unit="1")={0,1,0}
+    input Real widthDirection[3]={0,1,0}
```
Comment (tobolar) [inline Modelica/Utilities/Internal.mo:26]: The change is fine. And I would even addionally set `lengthDirection` and `widthDirection` being of `SI.Position` type.
@MartinOtter what do you mean?

**Review by tobolar — APPROVED** [review]
Fine to me.

File: Modelica/Utilities/Internal.mo
Comment (tobolar) [inline Modelica/Utilities/Internal.mo:1]: Consequently, also the unit of `Modelica.Mechanics.MultiBody.Types.Axis` shall be deleted. It is used e.g. in `Modelica.Mechanics.MultiBody.Visualizers.FixedShape`

Comment (HansOlsson) [reply Modelica/Utilities/Internal.mo:1]: Oh, that is used all over the place. Will have to check that it doesn't impact anything else.

**Comment by Harisankar-Allimangalath** [comment]
@MartinOtter can you please review the changes and see if it is enough . Thankyou

**Comment by tobolar** [comment]
@Harisankar-Allimangalath I thing https://github.com/modelica/ModelicaStandardLibrary/pull/4194#pullrequestreview-1630946395 shall be resolved first.

**Review by MartinOtter — APPROVED** [review]

**Review by MartinOtter — APPROVED** [review]

**Pull request merged**